File: be_hotel/src/main/java/be_hotel/serive/booking/BookingService.java
package be_hotel.serive.booking;

import be_hotel.Modal.entity.Booking;
import be_hotel.Modal.entity.Room;
import be_hotel.Modal.request.BookingFilterRequest;
import be_hotel.Modal.request.BookingRequest;
import org.springframework.data.domain.Page;

import java.time.LocalDateTime;

public interface BookingService {
    public Page<Booking> getAllBooking(int page, int size);

    public Booking updateBooking(BookingRequest bookingRequest, Integer id);

    public String deleteBooking(Integer id);

    public boolean isRoomAvailable(BookingFilterRequest bookingFilterRequest);


    public Booking createBooking(BookingRequest bookingRequest);
}

File: src/services/BookingService.js
import api from "./api";

const BookingService = {
    getBooking(page, size){
        return api.get(`getAllBooking/${page}/${size}`)
    },
    createBooking(body){
        return api.post("createBooking", body)
    },
    updateBooking(id, body){
        return api.put("updateBooking/"+id, body)
    },
    deleteBooking(id){
        return api.delete("deleteBooking/"+id)
    }
}

export default BookingService;